py/stackctrl.c

```diff
@@ -28,8 +28,10 @@
 #include "py/stackctrl.h"
 
 void mp_stack_ctrl_init(void) {
+    #ifndef __clang_analyzer__ // stack_dummy is an intentional dangling reference
     volatile int stack_dummy;
     MP_STATE_THREAD(stack_top) = (char *)&stack_dummy;
+    #endif
```

Comment:
I don't think this is a good idea, to remove this code when the analyzer is enabled.  Is there another way to somehow silence the warning?

Reply:
I could not find a way. But you're right, I'll remove this change not to add any regression while we found a better way to silent the warning.